app/connectors/OttConnector.scala

```diff
@@ -42,38 +47,113 @@ class OttConnector @Inject() (config: Configuration, httpClient: HttpClientV2)(i
   private def ottCountriesUrl =
     url"$baseUrl/xi/api/v2/geographical_areas/countries"
 
-  private def getFromOtt[T](url: URL, authToken: String)(implicit
+  private def getFromOtt[T](
+    url: URL,
+    authToken: String,
+    auditDetails: Option[OttAuditData]
+  )(implicit
     hc: HeaderCarrier,
     reads: Reads[T]
   ): Future[T] = {
     val newHeaderCarrier = hc.copy(authorization = Some(Authorization(authToken)))
 
+    val requestStartTime = Instant.now
+
     httpClient
       .get(url)(newHeaderCarrier)
       .execute[HttpResponse]
       .flatMap { response =>
+        val requestEndTime = Instant.now
+
         response.json
           .validate[T]
-          .map(result => Future.successful(result))
+          .map { result =>
+            auditService.auditOttCall(
+              auditDetails,
+              requestStartTime,
+              requestEndTime,
+              response.status,
+              None,
+              Some(result)
+            )
+            Future.successful(result)
+          }
           .recoverTotal(error => Future.failed(JsResult.Exception(error)))
       }
-      .recoverWith { case e: NotFoundException =>
-        Future.failed(UpstreamErrorResponse(e.message, NOT_FOUND))
+      .recoverWith {
+        case e: HttpException =>
+          auditService.auditOttCall(auditDetails, requestStartTime, Instant.now, e.responseCode, Some(e.message), None)
+
+          Future.failed(UpstreamErrorResponse(e.message, e.responseCode))
+
+        case e: UpstreamErrorResponse =>
+          auditService.auditOttCall(auditDetails, requestStartTime, Instant.now, e.statusCode, Some(e.message), None)
+          Future.failed(e)
+
+        case e: Exception =>
+          //E.g. Any error not directly related to the http call, e.g. Json parsing
+          auditService.auditOttCall(auditDetails, requestStartTime, Instant.now, OK, Some(e.getMessage), None)
```

Comment:
I'm not sure about auditing in this case, the status OK doesn't seem to make sense here

Reply:
I realised this case was only really intended to catch that error we throw if we got back a successful OK from the server but couldn't parse the json. So I've just moved the audit to there and ditched this case.

Reply:
Awesome :)